Short-Answer/Algorithms_Answers.md

```diff
@@ -0,0 +1,37 @@
+Add your answers to the Algorithms exercises here.
+
+a) O(n)
+
+b) O(n^2log(n))
```

Comment:
The `j` and `k` loops don't go all the way to `n`, but they go some fraction of the way to `n`. And fractional constants are dropped.

The `l` loop is tricky. Close examination shows that it is:

```
for (l = 1 + k; l < 10 + k; l++)
```

but `k` doesn't change throughout a single loop of `l`. So it's effectively constant and the loop is equivalent to:

```
for (l = 1; l < 10; l++)     O(10) == O(1)
```

No matter what `k` you plug into that loop, it always cycles 10 times.

Since the loops are all nested, we multiply the values.

```
O(1) +  <-- this is for sum = 0
O(n) * O(n) * O (n) * O(1) <-- for nested loops

Simplifying:

O(1) + O(n*n*n*1)
O(1) + O(n^3)
O(n^3)               O(n^3) dominates
```

Answer: `O(n^3)`